pkg/runtime/pubsub/subscription.go

```diff
@@ -2,6 +2,26 @@ package pubsub
 
 import "fmt"
 
+type SubscriptionType int32
+
+const (
+	SubscriptionTypeUnknown = iota
+	SubscriptionTypeDeclarative
+	SubscriptionTypeProgrammatic
+	SubscriptionTypeStreaming
+)
+
+var subscriptionTypeName = map[int32]string{
```

Comment:
Can you make this a `switch` inside the `String()` method? This way we avoid allocating a map in the global scope

Reply:
actually changed it to add it to the proto of subscription because that is how it was used in the metadata in api/universal.